### PR TITLE
Fix operational semantics

### PR DESCRIPTION
**Pull request by jaredhoberock**
This change was prompted by Chris M's recent email

### CODE REVIEW

**Comment by jaredhoberock** [comment]
@ccmysen Do these changes look like they address the issue you raised over exception propagation? Do you know of any other wording where the paper says that potentially asynchronous functions have to throw exceptions?

File: README.md [hunk 1]
```diff
@@ -651,7 +651,7 @@ In the Table below, `x` denotes a (possibly const) executor object of type `X`,
 
 | Expression | Return Type | Operational semantics |
 |------------|-------------|---------------------- |
-| `x.'e'(...)` <br/> `'e'(x, ...)` | void | Throws any exception thrown by `f()`. |
+| `x.'e'(...)` <br/> `'e'(x, ...)` | void | |
```
Comment (chriskohlhoff) [inline README.md]: Should we say something like (maybe a non-normative note): If f() exits via an exception, the behavior is specific to the executor?

Comment (jaredhoberock) [inline README.md]: Good idea, I'll add a note.

File: README.md [hunk 2]
```diff
@@ -667,8 +667,8 @@ In the Table below, `x` denotes a (possibly const) executor object of type `X`,
 
 | Expression | Return Type | Operational semantics |
 |------------|-------------|---------------------- |
-| `x.async_'e'(...)` <br/> `async_'e'(x, ...)` | A type that satisfies the `Future` requirements for the value type `R`. |  Throws any exception thrown by `f()`. |
-| `x.then_'e'(..., pred, ...)` <br/> `then_'e'(x, ..., pred, ...)` | A type that satisfies the `Future` requirements for the value type `R`. | Throws any exception thrown by `f()`. |
+| `x.async_'e'(...)` <br/> `async_'e'(x, ...)` | A type that satisfies the `Future` requirements for the value type `R`. |  Asynchronously returns the result of `f()` or any exception thrown by `f()` via the resulting `Future`. |
+| `x.then_'e'(..., pred, ...)` <br/> `then_'e'(x, ..., pred, ...)` | A type that satisfies the `Future` requirements for the value type `R`. | Asynchronously returns the result of `f(pr)` or any exception thrown by `f(pr)` via the resulting `Future`. |
```
Comment (chriskohlhoff) [inline README.md]: I think maybe it should be: Stores the result of `f(pr)`, or any exception thrown by `f(pr)`, in the associated shared state of the resulting `Future`. When we flesh out the Future requirements it should probably be done using terms like "associated shared state", etc.

**Comment by chriskohlhoff** [comment]
This looks good to me now. Ready to merge?

**Comment by jaredhoberock** [comment]
Yep, merged.

**Pull request merged**